compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -591,6 +591,63 @@ TR::Register *OMR::X86::TreeEvaluator::fpBinaryArithmeticEvaluator(TR::Node
       }
    }
 
+TR::Register *OMR::X86::TreeEvaluator::fpAbsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   static uint8_t MASKOFFLOAT[] =
+      {
+      0xff, 0xff, 0xff, 0x7f,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      };
+   static uint8_t MASKOFDOUBLE[] =
+      {
+      0xff, 0xff, 0xff, 0xff,
+      0xff, 0xff, 0xff, 0x7f,
+      0x00, 0x00, 0x00, 0x00,
+      0x00, 0x00, 0x00, 0x00,
+      };
+   auto mask = MASKOFDOUBLE;
+   auto child = node->getFirstChild();
+   auto value = cg->evaluate(child);
+   auto result = cg->allocateRegister(value->getKind());
+
+   if (value->isSinglePrecision())
+      {
+      result->setIsSinglePrecision();
+      mask = MASKOFFLOAT;
+      }
+
+   if (value->getKind() != TR_FPR) // Legacy supported for X87, to be deleted
```

Comment:
X87 is current in-use by OMR on x86-32, although is not used by any known downstream project. All current OMR tests on X86-32 uses the x87 path. It may be too early to remove this path before OMR completely drops x87.